### PR TITLE
Add unit tests for NAS

### PR DESCRIPTION
**Pull request by ultmaster**

### CODE REVIEW

File: src/sdk/pynni/nni/nas/pytorch/darts/mutator.py [hunk 1]
```diff
@@ -63,18 +63,21 @@ def sample_final(self):
...
+                    for src_key in mutable.choose_from:
+                        if src_key not in edges_max:
+                            _logger.warning("InputChoice.NO_KEY in '%s' is weighted 0 when selecting inputs.", mutable.key)
+                        weights.append(edges_max.get(src_key, 0.))
+                    weights = torch.tensor(weights)  # pylint: disable=not-callable
+                    _, topk_edge_indices = torch.topk(weights, mutable.n_chosen)
+                    selected_multihot = []
+                    for i, src_key in enumerate(mutable.choose_from):
+                        if i not in topk_edge_indices and src_key in result:
+                            result[src_key] = torch.zeros_like(result[src_key])  # clear this choice to optimize calc graph
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/nas/pytorch/darts/mutator.py]: what is the meaning of this line?

File: src/sdk/pynni/nni/nas/pytorch/utils.py [hunk 2]
```diff
@@ -17,6 +17,14 @@ def global_mutable_counting():
     return _counter
 
 
+def reset_global_mutable_counting():
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/nas/pytorch/utils.py]: global counting is still here?

File: src/sdk/pynni/nni/nas/pytorch/darts/mutator.py (hunk 1 above)
Comment (ultmaster) [reply src/sdk/pynni/nni/nas/pytorch/darts/mutator.py]: If an edge is never selected, there is no need to calculate any op on this edge. This is to eliminate redundant calculation. If there is any problem with this, open an issue.

File: src/sdk/pynni/nni/nas/pytorch/utils.py (hunk 2 above)
Comment (ultmaster) [reply src/sdk/pynni/nni/nas/pytorch/utils.py]: Changed to private method for future refactor (remove global counting entirely)

Comment (QuanluZhang) [reply src/sdk/pynni/nni/nas/pytorch/utils.py]: ok

File: src/sdk/pynni/nni/nas/pytorch/darts/mutator.py (hunk 1 above)
Comment (QuanluZhang) [reply src/sdk/pynni/nni/nas/pytorch/darts/mutator.py]: please add comments here to explain.

**Pull request merged**